src/Security/Authorization/Policy/src/AuthorizationEndpointConventionBuilderExtensions.cs

```diff
@@ -143,6 +143,24 @@ public static TBuilder RequireAuthorization<TBuilder>(this TBuilder builder, Act
         return builder;
     }
 
+    /// <summary>
+    /// Enables AuthorizationMiddleware to cache the policy for a particular endpoint so its only computed once when using the default DefaultAuthorizationPolicyProvider.
+    /// </summary>
+    /// <param name="builder">The endpoint convention builder.</param>
+    /// <returns>The original convention builder parameter.</returns>
+    public static TBuilder WithAuthorizationCache<TBuilder>(this TBuilder builder) where TBuilder : IEndpointConventionBuilder
+    {
+        builder.Add(endpointBuilder =>
```

Comment:
How about to only add `AuthorizationPolicyCache` if there are `IAuthorizeData` or `AuthorizationPolicy` is available in the metadata. Also the new `Finally` method could be used instead of `Add`.